src/components/transaction.mjs

```diff
@@ -0,0 +1,27 @@
+// @flow
+
+import crypto from 'crypto';
+
+import type Account from './account';
+
+class Transaction {
+  id: string;
+  amount: number;
+  sender: Account;
+  receiver: Account;
+  timestamp: number;
+
+  constructor(amount: number, from: Account, to: Account) {
+    this.id = crypto.randomBytes(32).toString('hex');
+    this.amount = amount;
+    this.sender = from;
+    this.receiver = to;
+    this.timestamp = Date.now();
+  }
+
+  async verify(): Promise<boolean> {
+    return this.sender.balance >= this.amount;
```

Comment:
Throws error if amount <= 0

Reply:
The error could be handled elsewhere, so the corresponding test might currently passes. It's better to  raise the error and even prevent the transaction from being created in the first place.

Reply:
Why would it throw error if this.amount <= 0 ? It only returns true if balance >= amount, and false if not. Returning a boolean allows us to do smth like `if (tx.verify()) {/* ... */}`